new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSecureNodePort.java

```diff
@@ -0,0 +1,313 @@
+  private static String wlsBaseImage = WLS_BASE_IMAGE_NAME + ":" + WLS_BASE_IMAGE_TAG;
+  private static boolean isUseSecret = true;
+  private static int replicaCount = 2;
+  private static final String domainUid = "mii-default-secure";
+  private StringBuffer curlString = null;
+
+  private StringBuffer checkCluster = null;
+  private static Map<String, Object> secretNameMap;
+
+  private final String adminServerPodName = domainUid + "-admin-server";
+  private final String managedServerPrefix = domainUid + "-managed-server";
+  private final String adminServerName = "admin-server";
+
+  private static LoggingFacade logger = null;
+
+  /**
+   * Install Operator.
+   * Create domain resource definition.
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = getLogger();
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+        .and().with().pollInterval(10, SECONDS)
+        .atMost(5, MINUTES).await();
+
+    // get a new unique opNamespace
+    logger.info("Creating unique namespace for Operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    logger.info("Creating unique namespace for Domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domainNamespace = namespaces.get(1);
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // Create the repo secret to pull the image
+    assertDoesNotThrow(() -> createDockerRegistrySecret(domainNamespace),
+          String.format("createSecret failed for %s", REPO_SECRET_NAME));
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    String adminSecretName = "weblogic-credentials";
+    assertDoesNotThrow(() -> createDomainSecret(adminSecretName,"weblogic",
+            "welcome1", domainNamespace),
+            String.format("createSecret failed for %s", adminSecretName));
+
+    // create encryption secret
+    logger.info("Create encryption secret");
+    String encryptionSecretName = "encryptionsecret";
+    assertDoesNotThrow(() -> createDomainSecret(encryptionSecretName, 
+            "weblogicenc", "weblogicenc", domainNamespace),
+             String.format("createSecret failed for %s", encryptionSecretName));
+
+    String configMapName = "default-secure-configmap";
+    String yamlString = "topology:\n"
+        + "  Server:\n"
+        + "    'admin-server':\n"
+        + "       SSL: \n"
+        + "         Enabled: true \n"
+        + "         ListenPort: '7008' \n";
+    createModelConfigMap(configMapName, yamlString);
+
+    //determine if the tests are running in Kind cluster. if true use images from Kind registry
```

Comment:
where are you using this wlsBaseImage image?

Reply:
These are boiler plate generic code from existing Mii Domain Class.  I am not building any domain images in this class. The ImageBuilder class will build the domain image by default.